### PR TITLE
Signaller refactor

### PR DESCRIPTION
**Pull request by harrynull**
Move `signaller` to `Capturer`. Also rework how signaller passes messages.

### CODE REVIEW

File: src/gui/component/sharing.rs [hunk 1]
```diff
@@ -92,33 +97,37 @@ impl<'a> Component<'a> for SharingPage {
...
             action_bar(),
-        ].align_items(Center)
-            .width(Fill)
-            .height(Fill)
-            .into()
+        ]
+        .align_items(Center)
+        .width(Fill)
+        .height(Fill)
+        .into()
```
Comment (HITGIF) [inline src/gui/component/sharing.rs:119]: Is this from you or your formatter? My formatter isn't working for these...

File: src/signaller/websocket_signaller.rs [hunk 2]
```diff
@@ -1,163 +1,180 @@
...
+        let $topic = $self.topics_rx
+                                                                  .read()
+                                                                  .await
+                                                                  .get($discriminant.into())
+                                                                  .unwrap()
+                                                                  .lock()
+                                                                  .await
+                                                                  .recv()
+                                                                  .await
+                                                                  .unwrap() else { unreachable!() };
```
Comment (HITGIF) [inline src/signaller/websocket_signaller.rs:146]: too much indent?

File: src/capture/capturer.rs [hunk 3]
```diff
@@ -37,32 +38,45 @@ pub struct Capturer {
     pub args: Args,
     pub config: Config,
     shutdown_token_opt: Option<CancellationToken>,
-    invite_link_opt: Option<String>,
-    room_id_opt: Option<String>,
+    signaller: Arc<Mutex<Option<Arc<dyn Signaller + Send + Sync>>>>,
+    state_update_callback: Arc<dyn Fn() + Send + Sync>,
```
Comment (HITGIF) [inline src/capture/capturer.rs:42]: nit: notify_update

File: src/gui/component/sharing.rs (hunk 1 above)
Comment (harrynull) [reply src/gui/component/sharing.rs:119]: Run `cargo clippy --fix` and `cargo fmt`

**Review by HITGIF — APPROVED** [review]

**Pull request merged**